src/fdcache.cpp

```diff
@@ -843,6 +870,10 @@ bool PageList::Serialize(CacheFileStat& file, bool is_output)
     }
 
     string strall = ssall.str();
+    if(-1 == ftruncate(file.GetFd(), strall.length())){
```

Comment:
Would it be safer to truncate to zero?  This would ensure that if the subsequent write (partially?) fails that there is not garbage in the file.

Reply:
Actually it seems like we should probably write to a temp file and rename onto the existing cache file for atomicity reasons.  What do you think?

Reply:
I understand, I will change to truncate the file size to 0.
And I will check the source code again for the operation linked with the cache file.
please wait a moment.

Reply:
I modified the source code so that the size of truncate is 0.
I also comment outside of this conversation about this PageList file conflict.
Then, apply for the correction in another PR different from this PR.